src/domain/alerts/alerts.repository.ts

```diff
@@ -183,19 +192,43 @@ export class AlertsRepository implements IAlertsRepository {
       this.loggingService.debug(
         `An alert log for an transaction with no verified emails associated was thrown for Safe ${args.newSafeState.address}`,
       );
-    } else {
-      return this.emailApi.createMessage({
-        to: emails,
-        template: this.configurationService.getOrThrow<string>(
-          'email.templates.recoveryTx',
-        ),
-        // TODO: subject and substitutions need to be set according to the template design
-        subject: 'Recovery attempt',
-        substitutions: {
-          owners: JSON.stringify(args.newSafeState.owners),
-          threshold: args.newSafeState.threshold.toString(),
-        },
-      });
+      return;
     }
+
+    return this.emailApi.createMessage({
+      to: emails,
+      template: this.configurationService.getOrThrow<string>(
+        'email.templates.recoveryTx',
+      ),
+      subject: AlertsRepository.RECOVERY_TX_EMAIL_SUBJECT,
+      substitutions: {
+        chainId: args.chainId,
+        safeAddress: this._formatAddress(args.newSafeState.address),
+        // PushWoosh doesn't allow substitution arrays so we send pre-structured owners element
+        owners: `<ul>${args.newSafeState.owners.map(
+          (owner) => `<li>${this._formatAddress(owner)}</li>`,
+        )}</ul>`,
+        threshold: args.newSafeState.threshold.toString(),
+      },
+    });
+  }
+
+  /**
+   * Checksums and formats the address to be displayed in the email as
+   * PushWoosh doesn't support JavaScript
+   *
+   * 0x1234...7890 => 0x1234<span id="address-center">.../span>7890
+   * @param address - address to be formatted
+   */
+  private _formatAddress(address: string): string {
```

Comment:
I'd say the `domain` layer is the place to include provider-specific (PushWoosh in this case) logic.

Maybe we could left it here in a first iteration, but in my opinion it would make more sense to have something like a provider-specific templates service (something like `pushwoosh-templates.service.ts` in this case) in the `datasources` layer, following a common service interface that could contain functions like `singleAddressToHtml` and `addressListToHtml` for example, so in the future another emailing provider would need to implement that interface, and the `domain` layer would stay provider-agnostic.

As I said, maybe could be seen as a follow-up instead of a matter for this PR. Thoughts? @fmrsabino @iamacook 🙂

Reply:
Good point and I agree. I will create the appropriate service now.

Reply:
I adjusted the structure in 553dd9c8dc0351d651beeecc5de4cac0c096e549 and updated test coverage in 2df8bbc0d64e06431628a953c9a5f18d42bb19dd..